toy_settings/django_back_end/models.py

```diff
@@ -25,3 +25,14 @@ class Event(models.Model):
     timestamp = models.DateTimeField()
     payload = models.CharField(max_length=500)
     payload_converter = cattrs.preconf.json.make_converter()
+
+
+class Sequence(models.Model):
+    event = models.ForeignKey(Event, on_delete=models.PROTECT)
+    key = models.CharField(max_length=100)
+    idx = models.PositiveIntegerField()
+
+    class Meta:
+        constraints = [
+            models.UniqueConstraint(fields=["key", "idx"], name="unique_idx_per_key")
+        ]
```

Comment:
Thought: On a non-toy system we'd want some indexes here for the common lookups (especially the one used by the trigger).

Reply:
Good point -- i assume just an index on the key in both tables?